homeassistant/components/recorder/__init__.py

```diff
@@ -230,8 +230,8 @@ def purge_ticker(event):
                 self.queue.task_done()
                 continue
 
-            entity_id = event.data.get(ATTR_ENTITY_ID)
-            domain = event.data.get(ATTR_DOMAIN)
+            entity_id = str(event.data.get(ATTR_ENTITY_ID))
```

Comment:
This is always a string? If not, we should error out instead of trying to "fix" it because it means malformed state changed events are being sent.